### PR TITLE
style: Move outline-style outside of mako

### LINKED ISSUE #19546: Move outline-style outside of mako.
I'll take this one as a sub-issue of #19015

### PR DESCRIPTION
**Pull request by CYBAI**
This is a sub-PR of #19015 
r? emilio 

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #19546 
- [x] These changes do not require tests

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19548)

### CODE REVIEW

File: components/style/values/computed/outline.rs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
```
Comment (tigercosmos) [inline components/style/values/computed/outline.rs]: `*/` should move to next line?

File: components/style/values/computed/outline.rs [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+//! Computed values for outline properties
+
+pub use values::specified::OutlineStyle;
```
Comment (tigercosmos) [inline components/style/values/computed/outline.rs]: need a new line at the end

Comment (CYBAI) [reply components/style/values/computed/outline.rs]: Fixed! Thanks.

File: components/style/values/computed/outline.rs (hunk 1 above)
Comment (CYBAI) [reply components/style/values/computed/outline.rs]: I think maybe it's okay.
I just copied this License from other files.

File: components/style/values/specified/outline.rs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    /// Get value for None or Hidden
+    pub fn none_or_hidden(&self) -> bool {
+        match *self {
+            Either::First(ref _auto) => false,
+            Either::Second(ref border_style) => border_style.none_or_hidden()
+        }
+    }
+
+    /// Parse outline-style
+    pub fn parse<'i, 't>(
```
Comment (emilio) [inline components/style/values/specified/outline.rs]: Hmm, so this is weird, how are we parsing `auto` now?

Before that, we used `Parse`, and then reject `BorderStyle::Hidden`, now that's not happening.

Let's change `OutlineStyle` to not be `Either`, but just be:

```rust
#[derive(ToCss, ToComputedValue, ...)]
pub enum OutlineStyle {
    Auto,
    Other(BorderStyle),
}
```

And parse `Auto` manually.

File: components/style/values/specified/outline.rs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    ) -> Result<OutlineStyle, ParseError<'i>> {
+        if let Ok(result) = input.try(|i| BorderStyle::parse(context, i)) {
+            match result {
+                BorderStyle::Hidden => {
+                    Err(input.new_custom_error(SelectorParseErrorKind::UnexpectedIdent("hidden".into())))
+                },
+                _ => Ok(Either::Second(result))
+            }
+        } else {
+            Err(input.new_custom_error(SelectorParseErrorKind::UnexpectedIdent("hidden".into())))
```
Comment (emilio) [inline components/style/values/specified/outline.rs]: This is wrong, this is returning an "unexpected token: hidden" error, when `BorderStyle` doesn't parse.

Instead we should propagate the error up.

So, with the above, it'd look something like:

```rust
if let Ok(border_style) = input.try(|i| BorderStyle::parse(context, i)) {
    if matches!(border_style, BorderStyle::Hidden) {
        return Err(input.new_custom_error(SelectorParseErrorKind::UnexpectedIdent("hidden".into()))) 
    }
    return Ok(OutlineStyle::Other(border_style));
}

input.expect_ident_matching("auto")?;
Ok(OutlineStyle::Auto)
```

File: components/style/values/specified/outline.rs (hunk 3 above)
Comment (CYBAI) [reply components/style/values/specified/outline.rs]: Thanks!
I think using `enum` is more reasonable for this attribute.
I'm going to try it.

**Review by emilio — APPROVED** [review]
Looks great, thanks @CYBAI!

**Pull request merged**